src/Raven.Studio/typescript/viewmodels/database/patch/patch.ts

```diff
@@ -364,12 +367,50 @@ class patch extends viewModelBase {
         return type !== "unknown" ? patch.recentKeyword + " (" + collectionIndexName + ")" : patch.recentKeyword;
     }
 
-    private patchOnQuery() {
+    private getMatchingDocumentsNumber(): JQueryPromise<number> {
+        const patchScript = this.patchDocument().query();
+        const patchScriptParts = patchScript.split("update");
+        
+        const matchingDocs = $.Deferred<number>();
+        
+        if (patchScriptParts.length === 2) {
+            let query = queryCriteria.empty();
+            query.queryText(patchScriptParts[0]);
+
+            new queryCommand(this.activeDatabase(), 0, 0, query)
+                .execute()
+                .done((queryResults: pagedResultExtended<document>) => matchingDocs.resolve(queryResults.totalResultCount))
+                .fail(() => matchingDocs.resolve(-1))
+        } else {
+            matchingDocs.resolve(-1);
+        }
+
+        return matchingDocs;
+    }
+    
+    private executePatch(matchingDocuments: number): void {
         eventsCollector.default.reportEvent("patch", "run");
-        const message = `Are you sure you want to apply this patch to matching documents?`;
 
-        this.confirmationMessage("Patch", message, {
-            buttons: ["Cancel", "Patch all"]
+        const patchQuestion = `<div>Are you sure you want to apply this patch to matching documents?</div>`;
+        
+        const warningMessage = `<li>
+                                 <small>Actual number of processed documents might be smaller if documents are filtered by the 'update' script</small>
+                             </li>`;
+        
+        const patchMessage = matchingDocuments > -1 ?
+                             `<div class="margin-bottom margin-bottom-lg text-info bg-info padding padding-xs">
+                                 <ul class="margin-top">
+                                     <li>
+                                         <small>Number of documents matching the Patch Query: <strong class="margin-left margin-left-sm">${matchingDocuments}</strong></small>
```

Comment:
format number to avoid count = `12334234`

Reply:
done